File: c/brutprim.c
/*
 * brutprim.c: The world's worst prime number generator.
 */

#include <stdio.h>

int main() 
{
    int l,n,d,p;

    printf("LIMIT ? ");
    scanf("%d", &l);

    for (n=3; n<=l; n++) {
        p=1;
        for (d=2; d<=(n-1); d++) {
            if ((n % d)==0) {
                putchar('.');
                p=0;
                break;
            }
        }
        if (p) {
            printf("%d ", n);
        }
    }
    printf("\n");
} 

File: forth/brutprim.f
: banner ." Brutprim: the worlds worst prime number generator. www.smbaker.com" CR ;
: check 2 BEGIN 2DUP = IF DROP . EXIT THEN 2DUP MOD 0= IF 46 EMIT DROP DROP EXIT THEN 1+ AGAIN ;
: brutprim banner 255 3 DO I check LOOP ;
